src/utils/auth/create.new.user.ts

```diff
@@ -0,0 +1,46 @@
+import prismaClient from "../../client.js";
+import crypto, { pbkdf2Sync } from "crypto";
+
+export async function createNewUser(username: string, password: string) {
+	const passwordHashAndSalt = await hashPassword(password);
+
+	const createdUser = await prismaClient.user.create({
+		data: {
+			username,
+			passwordHash: passwordHashAndSalt.hashedPassword,
+			salt: passwordHashAndSalt.salt,
+			role: "user",
+		},
+	});
+
+	if (createdUser.id) {
+		return createdUser;
+	} else {
+		throw new Error("CREATE_USER_FAILED");
+	}
+}
```

Comment:
Always use early returns. It makes the code cleaner
```suggestion
	if (!createdUser.id) throw new Error("CREATE_USER_FAILED");
	
	return createdUser;
}
```